### PR TITLE
Fix for HFA args on ARM32

### PR DESCRIPTION
**Pull request by briansull**
Test fix for issue #5326

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@parjong  @myungjoo PTAL
@kyulee1 @CarolEidt PTAL

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -4019,6 +4019,9 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
                 }
                 else
                 {
+                    // Currently the only non-HFA multireg struct is on ARM64 
+                    // and is two registers in size (i.e. two slots)
+                    assert(varDsc->lvSize() == (2 * TARGET_POINTER_SIZE));
```
Comment (CarolEidt) [inline src/jit/codegencommon.cpp]: This comment is not correct. We also have non-Hfa multireg structs on amd64/ux, and we have longs on 32-bit architectures. As it happens, they are always two registers. But the comment is misleading.

File: src/jit/codegencommon.cpp [hunk 2]
```diff
@@ -4361,16 +4364,22 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
...
+#ifdef _TARGET_ARM_
+                // On ARM32 the storeType for HFA args is always TYP_FLOAT
+                storeType = TYP_FLOAT;
+                slotSize = (unsigned)emitActualTypeSize(storeType);
+#else   // _TARGET_ARM64_
                 storeType = genActualType(varDsc->GetHfaType());
-                slotSize  = (unsigned) emitActualTypeSize(storeType);
+                slotSize = (unsigned)emitActualTypeSize(storeType);
+#endif // _TARGET_ARM64_
             }
```
Comment (parjong) [inline src/jit/codegencommon.cpp]: When I tested only with these changes, there was another assertion failure. Here is the assertion failure that I encountered.
```
Assert failure(PID 24305 [0x00005ef1], Thread: 24305 [0x5ef1]): Assertion failed 'regArgMaskLiveSave != regArgMaskLive' in 'DomainBoundILStubClass:IL_STUB_PInvoke(struct):struct' (IL size 40)

    File: /home/parjong/projects/toolchains/coreclr/src/jit/codegencommon.cpp Line: 4912
    Image: /mnt/dotnet-latest/Linux.arm.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 24305, Thread 24305/0x5ef1)' ****
  Assertion failed 'regArgMaskLiveSave != regArgMaskLive' in 'DomainBoundILStubClass:IL_STUB_PInvoke(struct):struct' (IL size 40)


/home/parjong/projects/toolchains/coreclr/src/jit/codegencommon.cpp, Line: 4912

Abort - Kill program
Retry - Debug
Ignore - Keep running
```
It seems that ``lvHfaSlots`` need to be revised in order to fix #5326.

File: src/jit/codegencommon.cpp (hunk 1 above)
Comment (briansull) [inline src/jit/codegencommon.cpp]: Yes, you are correct I will update this comment

File: src/jit/codegencommon.cpp (hunk 2 above)
Comment (briansull) [inline src/jit/codegencommon.cpp]: Yes it also needs the fix for ARM32 in lvHfaSlots

I will add it to this fix.

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@parjong @myungjoo PTAL
@kyulee1 @CarolEidt PTAL

I updated this with the additional fix for ARM32 in lvHfaSlots

**Comment by CarolEidt** [comment]
LGTM

**Comment by parjong** [comment]
LGTM 

When I checked, all the tests under JIT/jit64/hfa/main/testE are passed with this change.

**Comment by myungjoo** [comment]
LTGM. Thanks!

**Pull request merged**